contributors/devel/sig-architecture/api_changes.md

```diff
@@ -154,18 +159,42 @@ without changing the API version, so you can simply change it to:
 // Still API v6.
 type Frobber struct {
   Height int    `json:"height"`
-  Width  int    `json:"width"`
+  Width  *int   `json:"width"`  // optional, defaults to 42
   Param  string `json:"param"`
 }
 ```
 
+Note that fields which are added MUST be optional.  Optional fields must be
+nil-able (pointers or slices or (rarely) maps).
+
 The onus is on you to define a sane default value for `Width` such that rules
 #1 and #2 above are true - API calls and stored objects that used to work must
 continue to work.
 
+For newly added fields, the default behavior must be defined.  If this is a net
+new capability which did not exist before (i.e. a user must explicitly
+set it to change behavior), the default value should be "nothing" (`nil` in
+Go), meaning the feature is not activated. If this new field is extending an
+existing behavior, the default value should also be "nothing" (`nil`), meaning
+the existing behavior.  This satisfies rule #7.  If a user read the object
+before your feature and again after your feature, they would not see any
```

Comment:
I don't think we should guarantee this.

Reply:
That is the crux of this proposal.

I can say, as a consumer of another API wherein I want to run periodic audits to see if anything changed, I am presently very annoyed at EXACTLY this problem.  My audit keeps coming back with changes that are meaningless to me.

Reply:
SSA solves that problem, just apply with dry run and see if you would get any conflicts.

Reply:
That may be a viable alternative to this path.